databricks_cli/sdk/service.py

```diff
@@ -106,24 +161,42 @@ def submit_run(self, run_name=None, existing_cluster_id=None, new_cluster=None,
         if notebook_task is not None:
             _data['notebook_task'] = notebook_task
             if not isinstance(notebook_task, dict):
-                raise TypeError('Expected databricks.NotebookTask() or dict for field notebook_task')
+                raise TypeError(
+                    'Expected databricks.NotebookTask() or dict for field notebook_task'
+                )
         if spark_jar_task is not None:
             _data['spark_jar_task'] = spark_jar_task
             if not isinstance(spark_jar_task, dict):
-                raise TypeError('Expected databricks.SparkJarTask() or dict for field spark_jar_task')
+                raise TypeError(
+                    'Expected databricks.SparkJarTask() or dict for field spark_jar_task'
+                )
         if spark_python_task is not None:
             _data['spark_python_task'] = spark_python_task
             if not isinstance(spark_python_task, dict):
-                raise TypeError('Expected databricks.SparkPythonTask() or dict for field spark_python_task')
+                raise TypeError(
+                    'Expected databricks.SparkPythonTask() or dict for field spark_python_task'
+                )
         if spark_submit_task is not None:
             _data['spark_submit_task'] = spark_submit_task
             if not isinstance(spark_submit_task, dict):
-                raise TypeError('Expected databricks.SparkSubmitTask() or dict for field spark_submit_task')
+                raise TypeError(
+                    'Expected databricks.SparkSubmitTask() or dict for field spark_submit_task'
+                )
         if timeout_seconds is not None:
             _data['timeout_seconds'] = timeout_seconds
         if tasks is not None:
             _data['tasks'] = tasks
-        return self.client.perform_query('POST', '/jobs/runs/submit', data=_data, headers=headers, version=version)
+        if idempotency_token is not None:
+            _data['idempotency_token'] = idempotency_token
+        if job_clusters is not None:
+            _data['job_clusters'] = job_clusters
+        if git_source is not None:
+            _data['git_source'] = git_source
+            if not isinstance(git_source, dict):
+                raise TypeError('Expected databricks.GitSource() or dict for field git_source')
+        return self.client.perform_query(
+            'POST', '/jobs/runs/submit', data=_data, headers=headers, version=version
```

Comment:
`version` here should really be propagated as explicit path, e.g. `/api/2.1/jobs/runs/submit`...

Reply:
Perhaps, only that's not the way it's done in the CLI today.

I agree that we should have version scoped services. I.e. you instantiate the service object with the version, not define it on a per method basis. But that's a much larger refactor than should happen here. In fact, I don't think it's worth to do that work here but rather deal with it in vnext.